modin/experimental/torch/datasets.py

```diff
@@ -0,0 +1,108 @@
+from __future__ import annotations
+
+from typing import Hashable, NoReturn
+
+import numpy as np
+from pandas import DataFrame, Series
+from torch.utils import data as data_utils
+from torch.utils.data import DataLoader, IterableDataset
+
+from modin.pandas import DataFrame as ModinDataFrame
+from modin.pandas import Series as ModinSeries
+
+
+class ModinIterableDataset(IterableDataset):
+    """
+    PandasIterableDataset is responsible for converting a pandas/modin dataframe into a torch-compatible dataset.
+
+    The optional with_index option determines if index would be part of the output. Default is no.
+    """
+
+    def __init__(
+        self,
+        df: DataFrame | ModinDataFrame,
+        with_index: bool = False,
+    ) -> None:
+        """
+        Parameters
+        ----------
+
+        df: DataFrame
+            Anything that looks like a pandas dataframe. The data inside would be served row by row.
+
+        with_index: bool = False
+            Whether to include the index object with the result (like iterrows). Default is no.
```

Comment:
Formatting for consistency with the rest of code; check also what `scripts/doc_checker.py` reports for your changes.
```suggestion
        df : DataFrame | ModinDataFrame
            DataFrame-like object from which the iterable will draw data.

        with_index : bool, default: False
            If true, include the index object with the resulting iterable, similar to `DataFrame.iterrows`.
```